### PR TITLE
refact(operator): update lvm operator with custom priorityclass

### PR DESCRIPTION
**Pull request by prateekpandey14**
Signed-off-by: prateekpandey14 <prateekpandey14@gmail.com>

## Pull Request template

Please, go through these steps before you submit a PR.

**Why is this PR required? What issue does it fix?**:

**What this PR does?**:
 update zfs operator with custom priorityclass

**Does this PR require any upgrade changes?**:
No

Refer: #125
**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: deploy/lvm-operator.yaml [hunk 1]
```diff
@@ -1080,6 +1080,23 @@ roleRef:
...
+---
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: openebs-lvm-localpv-csi-controller-critical
+value: 900000000
+globalDefault: false
+description: "This priority class should be used for the OpenEBS LVM localPV CSI driver controller deployment only."
+---
+apiVersion: scheduling.k8s.io/v1
```
Comment (pawanpraka1) [inline deploy/lvm-operator.yaml:1092]: move this to node agent section below.

File: deploy/lvm-operator.yaml [hunk 2]
```diff
@@ -1114,7 +1131,7 @@ spec:
                 values:
                 - openebs-lvm-controller
             topologyKey: "kubernetes.io/hostname"
-      priorityClassName: system-cluster-critical
+      priorityClassName: openebs-lvm-localpv-csi-controller-critical
```
Comment (pawanpraka1) [inline deploy/lvm-operator.yaml:1134]: will it work for kube-system namespace? I remember there was some issue related to this.

File: deploy/lvm-operator.yaml (hunk 1 above)
Comment (prateekpandey14) [reply deploy/lvm-operator.yaml:1092]: Priorityclass should be exists before creating the sts using it.

File: deploy/lvm-operator.yaml (hunk 2 above)
Comment (prateekpandey14) [reply deploy/lvm-operator.yaml:1134]: Yes

Comment (prateekpandey14) [reply deploy/lvm-operator.yaml:1192]: I have tested locally once as well.

File: deploy/lvm-operator.yaml [hunk 3]
```diff
@@ -1147,6 +1147,14 @@ roleRef:
   name: openebs-lvm-provisioner-role
   apiGroup: rbac.authorization.k8s.io
 
+---
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: openebs-lvm-localpv-csi-controller-critical
+value: 900000000
```
Comment (pawanpraka1) [inline deploy/lvm-operator.yaml:1155]: what this values means?

Comment (prateekpandey14) [reply deploy/lvm-operator.yaml:1155]: A PriorityClass object can have any 32-bit integer value smaller than or equal to 1 billion. Larger numbers are reserved for critical system Pods that should not normally be preempted or evicted. The higher the value, the higher the priority.
Above is the max value we can set for custom priorityclass.

Comment (pawanpraka1) [reply deploy/lvm-operator.yaml:1155]: let us describe this in the comment.

Comment (prateekpandey14) [reply deploy/lvm-operator.yaml:1158]: added

**Review by pawanpraka1 — APPROVED** [review]
looks good.

**Pull request merged**